### ISSUE TITLE
Safari in private browsing mode throws error

### ISSUE THREAD

**Issue by i8ramin** [auth0/angular-storage#12]
Hey guys, it seems like Safari disables local/session storage when in private browsing mode. This causes angular apps to fail when trying to use the store methods. The error thrown:

Error: QuotaExceededError: DOM Exception 22

It would be nice to fall back to cookies in this instance or at least catch the error and do something with it.

**Comment by gimox** [comment]
+1 for the problem

please add this check for localStorage to resolve problem.
this work both Safari private mode and ios cordova

```javascript
function isLocalStorageSupported() {
  var storage = window. localStorage,
        key = 'test';
  try {
    storage.setItem(key, '1');
    storage.removeItem(key);
    return true;
  } catch (error) {
    return false;
  }
}
```

**Comment by mgonto** [comment]
Thanks for the report.

We'll implement it. I'll try to get it done this week, otherwise, next for sure :).

Thanks!

**Comment by mgonto** [comment]
I'll also check the PR first :). Thanks @i8ramin for the contributions :).

**Comment by gimox** [comment]
perfect!

**Comment by danmindru** [comment]
looking forward to the fix :+1: 
thanks.

**Comment by gimox** [comment]
i pushed a pull request, many time ago... since now...no merging... project dead?
if anyone is interested for a solution i made my trunk with fix https://github.com/gimox/angular-storage

**Comment by mgonto** [comment]
Fixed!

**Comment by gimox** [comment]
thanks

**Comment by mgonto** [comment]
And released as 0.0.10 :).

No prob!
